Jenkinsfile

```diff
@@ -1,11 +1,21 @@
 #!/bin/env groovy
 
-node {
-    stage ("Trigger") {
-        dir ("dlang/ci") {
-            git "https://github.com/dlang/ci.git"
-        }
+def cloneUpstream () {
+    checkout(scm: [
+        $class: 'GitSCM',
+        branches: scm.branches,
+        extensions: scm.extensions + [[$class: 'CleanBeforeCheckout']],
+        userRemoteConfigs: scm.userRemoteConfigs
+    ])
+}
 
-        load "dlang/ci/pipeline.groovy"
+def pipeline
+node {
+    dir('dlang/ci') {
+        cloneUpstream()
     }
+    pipeline = load 'dlang/ci/pipeline.groovy'
```

Comment:
Indeed it does, see https://ci.dlang.io/job/dlang-org/job/ci/job/PR-45/1/console, first takes Jenkinsfile from master, then pipeline.groovy from the PR commit.
This is good, as it actually tests the correct pipeline script of the PR.